.github/public-cloud/upload-image-to-azure.sh

```diff
@@ -0,0 +1,147 @@
+    echo "Error: AZURE_RESOURCE_GROUP, AZURE_STORAGE_ACCOUNT and AZURE_CONTAINER_NAME environment variables must be set."
+    exit 1
+  fi
+}
+
+LOCAL_VHD_PATH=$(readlink -f "$1")
+NAME=$(basename "$LOCAL_VHD_PATH" | sed 's/\.raw\.vhd$//') # just the file name without extension or path
+VERSION=$2
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+# shellcheck source=/dev/null
+source "$SCRIPT_DIR/cleanup-old-images-azure.sh"
+
+checkArguments "$@"
+checkEnvVars
+
+# === CHECK AZURE LOGIN ===
+az account show >/dev/null 2>&1 || { echo "Please run 'az login' first."; exit 1; }
+
+# === GET STORAGE ACCOUNT REGION ===
+echo "Fetching storage account region..."
+STORAGE_REGION=$(az storage account show --name "$AZURE_STORAGE_ACCOUNT" --query "primaryLocation" --output tsv)
+echo "Storage account is in region: $STORAGE_REGION"
+
+# === GET STORAGE ACCOUNT KEY ===
+echo "Fetching storage account key..."
+STORAGE_KEY=$(az storage account keys list --account-name "$AZURE_STORAGE_ACCOUNT" --query "[0].value" --output tsv)
+
+echo "Uploading VHD file ($LOCAL_VHD_PATH) to Azure Storage..."
+az storage blob upload --account-name "$AZURE_STORAGE_ACCOUNT" --container-name "$AZURE_CONTAINER_NAME" --type page \
+    --name "$NAME" --file "$LOCAL_VHD_PATH" --auth-mode key --account-key "$STORAGE_KEY" --overwrite true
+
+echo "Retrieving uploaded VHD URL..."
+VHD_URL=$(az storage blob url --account-name "$AZURE_STORAGE_ACCOUNT" --container-name "$AZURE_CONTAINER_NAME" --name "$NAME" --output tsv)
+echo "VHD uploaded successfully: $VHD_URL"
+
+# === Get file size in bytes ===
+VHD_SIZE_BYTES=$(stat -c %s "$LOCAL_VHD_PATH")
+
+# === Convert to GB and round up ===
+VHD_SIZE_GB=$(( (VHD_SIZE_BYTES + 1073741823) / 1073741824 ))
+echo "Calculated disk size: $VHD_SIZE_GB GB"
+
+echo "Creating a managed disk"
+az disk create \
+  --resource-group "$AZURE_RESOURCE_GROUP" \
+  --name "$NAME" \
+  --source "$VHD_URL" \
+  --os-type Linux \
+  --sku Premium_LRS \
+  --size-gb "$VHD_SIZE_GB" \
+  --hyper-v-generation V2
+
+echo "Getting the ID of the managed disk"
+DISK_ID=$(az disk show \
+  --resource-group "$AZURE_RESOURCE_GROUP" \
+  --name "$NAME" \
+  --query "id" \
+  --output tsv)
+
+# === CREATE AZURE IMAGE IN SAME REGION AS STORAGE ACCOUNT ===
+echo "Creating Azure image ($NAME) from VHD in region: $STORAGE_REGION..."
+az image create \
+	--resource-group "$AZURE_RESOURCE_GROUP" \
+	--name "$NAME" \
+	--os-type "Linux" \
+	--source "$DISK_ID" \
+	--hyper-v-generation "V2" \
+	--location "$STORAGE_REGION"
+echo "Image created successfully: $NAME"
+
+echo "Getting the image ID"
+IMAGE_ID=$(az image show \
+	--resource-group "$AZURE_RESOURCE_GROUP" \
+	--name "$NAME" \
+  --query "id" \
+  --output tsv)
+
+# echo "Creating a Shared Image Gallery (one-off)"
+# # https://learn.microsoft.com/en-us/azure/virtual-machines/create-gallery?tabs=portal%2Cportaldirect%2Ccli2
+# # TODO: Link to some EULA?
```

Comment:
This is an one-off creation of the image gallery. We can always edit the gallery details by hand in the web ui and change the link to the eula (as soon as we got one). It's mandatory field and I didn't know what to put here so I just linked to our Apache license in kairos repo.